src/data-access/typeORMDataAccess.ts

```diff
@@ -0,0 +1,120 @@
+import { DataSource } from 'typeorm';
+import * as dotenv from 'dotenv';
+import { User } from '../models/typeORMModels';
+import { UserUpdates } from '../types';
+
+dotenv.config();
+
+const username = process.env.DB_USERNAME || '';
+const DBname = process.env.NAME || '';
+const DBpassword = process.env.DB_PASSWORD || '';
+const DBhost = process.env.DB_HOST;
+
+export const AppDataSource = new DataSource({
+  type: 'postgres',
+  host: DBhost,
+  port: 5432,
+  username,
+  password: DBpassword,
+  database: DBname,
+  synchronize: true,
+  logging: true,
+  entities: [User],
+  subscribers: [],
+  migrations: [],
+});
+
+AppDataSource.initialize()
+  .then(() => {
+    console.log('Data Source has been initialized!');
+  })
+  .catch((err) => {
+    console.error('Error during Data Source initialization:', err);
+  });
+
+export async function createUser(newUser: User) {
+  try {
+    const res = await AppDataSource.transaction(
+      async (transactionalEntityManager) => {
+        const user = await transactionalEntityManager
+          .getRepository(User)
+          .create(newUser);
+        const results = await transactionalEntityManager
+          .getRepository(User)
+          .save(user);
+        return results;
+      }
+    );
+    return res;
+  } catch (error) {
+    console.error('Unable to connect to the database:', error);
```

Comment:
Well, you can't be sure that problem was in the connection. So the log may rather be vaguer, like, 'Database error'.

Reply:
fixed